pkg/instrumentation/auto/annotation.go

```diff
@@ -142,6 +73,33 @@ func (m *AnnotationMutators) Mutate(obj runtime.Object) bool {
 	}
 }
 
+func (m *AnnotationMutators) rangeObjectList(ctx context.Context, list client.ObjectList, option client.ListOption, fn objectCallbackFunc) {
+	if err := m.clientReader.List(ctx, list, option); err != nil {
+		m.logger.Error(err, "Unable to list objects",
+			"kind", fmt.Sprintf("%T", list),
+		)
+		return
+	}
+	switch l := list.(type) {
+	case *corev1.NamespaceList:
```

Comment:
I don't understand why have separate  cases here if we are doing the same functionality for all of them

Reply:
Same with this one. `client.ObjectList` doesn't expose the `Items` field, so there isn't a way to iterate over them without casting it.